pcsx2/PAD/Linux/wx_dialog/GamepadConfiguration.cpp

```diff
@@ -114,7 +114,7 @@ void GamepadConfiguration::OnSliderReleased(wxCommandEvent& event)
 
 		// convert in a float value between 0 and 1, and run rumble feedback.
 		// 0 to 1 scales to 0x0 to 0x7FFF
-		s_vgamePad[m_pad_id]->TestForce(m_sl_rumble_intensity->GetValue() / (float)0x7FFF);
```

Comment:
It would make more sense to change m_pad_id to be assigned `m_joy_map->GetSelection();` rather than removing it's use here. Or drop m_pad_id entirely.

Reply:
> It would make more sense to change m_pad_id to be assigned `m_joy_map->GetSelection();`

I think m_pad_id is not meant to be changed.
From what I understand, m_pad_id corresponds to a PS2 gamepad slot, so it can only be 0 or 1. The value is set when the window is created, based on which gamepad tab you have selected:
![image](https://user-images.githubusercontent.com/6643565/123523420-104baf00-d6cc-11eb-882f-7299ca745f3b.png)
This id (in the square brackets) is then used to update the selected gamepad in inis/PAD.ini: 
```ini
uid[0] = 5335467880578043658
uid[1] = 7874879626751023234
```

Please correct me if I'm wrong.

Reply:
@kenshen112 So, really the bug here: `s_vgamePad[m_pad_id]->TestForce(m_sl_rumble_intensity->GetValue() / (float)0x7FFF);` is that out of the vector of detected gamepads (s_vgamePad), it will always be the first or the second gamepad to vibrate, depending on which tab is selected.

Reply:
Ye kinda. Though m_pad_id is supposed to be the selected pad in general being that how it's currently updated or set is wrong. It would be more correct to have m_pad_id updated with the selected pad not the tab

Reply:
> m_pad_id is supposed to be the selected pad in general

What makes you think that? Because I think it's not.
If it were, then the widget would not know which PS2 pad to save the settings under in inis/PAD.ini, 0 or 1. 
m_pad_id basically only exists to do this, and is set only when the widget is created. 

The *selected* pad id is not stored, so it has to be retrieved with m_joy_map->GetSelection() for things like the force feedback test.

I don't see why anything would need to be changed here (unless I'm misunderstanding something about how the Linux pad GUI works)

Reply:
Mostly from the original usage. In TestForce, and that it's set to "pad" in the constructor. Mind you that's the tab but it was at least originally intended to be the pad index